### PR TITLE
fix(cx-14006): focus nested trigger on nested overlay close

### PR DESCRIPTION
**Pull request by neelabhnagaich**
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/mixins/FocusTrapMixin.ts [hunk 1]
```diff
@@ -279,6 +280,21 @@ export const FocusTrapMixin = <T extends AnyConstructor<FocusClass & FocusTrapCl
       }
     }
 
+    protected setFocusOnTrigger(triggerElement: HTMLElement) {
+      let deepNestedTriggerElement;
+      if (triggerElement.shadowRoot!) {
```
Comment (parsoman) [inline web-components/src/mixins/FocusTrapMixin.ts:285]: What will happen for a trigger element which is a span (non focusable element)? You can check the example from TPW customization button and check

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 2]
```diff
@@ -413,6 +414,10 @@ export namespace MenuOverlay {
       }
     }
 
+    private focusTheNestedTrigger(triggerEleement?: HTMLElement) {
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:417]: Function name can be focusOnTheNestedTrigger and correct the spelling of triggerElement .

File: web-components/src/mixins/FocusTrapMixin.ts [hunk 3]
```diff
@@ -279,6 +280,21 @@ export const FocusTrapMixin = <T extends AnyConstructor<FocusClass & FocusTrapCl
       }
     }
 
+    protected setFocusOnTrigger(triggerElement: HTMLElement) {
```
Comment (parsoman) [inline web-components/src/mixins/FocusTrapMixin.ts:283]: Please leave a comment to describe what this function actually achieve.

File: web-components/src/mixins/FocusTrapMixin.ts (hunk 1 above)
Comment (neelabhnagaich) [reply web-components/src/mixins/FocusTrapMixin.ts:285]: as the correct implementation of menu-overlay there needs to be a focusable element to work it properly, however we have handled the case when focusable element is inside a span then we are extracting it out.

File: web-components/src/mixins/FocusTrapMixin.ts (hunk 3 above)
Comment (neelabhnagaich) [reply web-components/src/mixins/FocusTrapMixin.ts:289]: done

File: web-components/src/components/menu-overlay/MenuOverlay.ts (hunk 2 above)
Comment (neelabhnagaich) [reply web-components/src/components/menu-overlay/MenuOverlay.ts]: done

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 4]
```diff
@@ -413,6 +414,10 @@ export namespace MenuOverlay {
       }
     }
 
+    private focusOnTheNestedTrigger(triggerElement?: HTMLElement) {
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:417]: Dont add "The".

Comment (neelabhnagaich) [reply web-components/src/components/menu-overlay/MenuOverlay.ts]: done

**Comment by neelabhnagaich** [comment]
BUILD

**Review by parsoman — APPROVED** [review]

**Review by merajcal — APPROVED** [review]

**Review by vinitkumargoel — APPROVED** [review]

**Pull request merged**